src/main/java/bio/overture/ego/security/CustomOAuth2UserInfoService.java

```diff
@@ -0,0 +1,132 @@
+package bio.overture.ego.security;
+
+import static java.util.Objects.isNull;
+
+import bio.overture.ego.model.enums.ProviderType;
+import bio.overture.ego.service.GithubService;
+import bio.overture.ego.service.LinkedinService;
+import bio.overture.ego.service.OrcidService;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class CustomOAuth2UserInfoService extends DefaultOAuth2UserService {
+
+  public static final String EMAIL = "email";
+  public static final String FAMILY_NAME = "family_name";
+  public static final String GIVEN_NAME = "given_name";
+  final GithubService githubService;
+  private final OrcidService orcidService;
+  private final LinkedinService linkedinService;
+
+  @Autowired
+  public CustomOAuth2UserInfoService(
+      GithubService githubService, OrcidService orcidService, LinkedinService linkedinService) {
+    this.githubService = githubService;
+    this.orcidService = orcidService;
+    this.linkedinService = linkedinService;
+  }
+
+  @Override
+  public CustomOAuth2User loadUser(OAuth2UserRequest oAuth2UserRequest)
+      throws OAuth2AuthenticationException {
+    OAuth2User oAuth2User = super.loadUser(oAuth2UserRequest);
+    try {
+      String provider = oAuth2UserRequest.getClientRegistration().getRegistrationId();
+      val idName = ProviderType.getIdAccessor(ProviderType.resolveProviderType(provider));
+      if (provider.equalsIgnoreCase(ProviderType.GITHUB.toString())) {
+        val info = getGithubUserEmail(oAuth2User, oAuth2UserRequest);
+        return CustomOAuth2User.builder()
+            .oauth2User(new DefaultOAuth2User(oAuth2User.getAuthorities(), info, idName))
+            .subjectId(info.getOrDefault(idName, "").toString())
```

Comment:
just checking, an empty `""` in this field would still result in an error and prevent the user being created in the db?

Reply:
yeah I think so will replace this with `get` instead of `getOrDefault` and make subjectId non nullable in the user class